[CQT-63] Update README (#254)
QuTech-Delft/OpenSquirrel: README.md

# OpenSquirrel

[![CI](https://github.com/QuTech-Delft/OpenSquirrel/workflows/Tests/badge.svg)](https://github.com/qutech-delft/OpenSquirrel/actions)
[![codecov](https://img.shields.io/codecov/c/github/QuTech-Delft/OpenSquirrel?style=flat-square&logo=codecov)](https://codecov.io/gh/QuTech-Delft/OpenSquirrel)
[![PyPI](https://badgen.net/pypi/v/OpenSquirrel)](https://pypi.org/project/OpenSquirrel/)
![OS](https://img.shields.io/badge/os-linux%20%7C%20macos%20%7C%20windows-blue?style=flat-square)
![Python](https://img.shields.io/badge/python-3.9%20%7C%203.10%20%7C%203.11%20%7C%203.12-blue.svg)
[![Code style: black](https://img.shields.io/badge/code%20style-black-000000.svg)](https://github.com/psf/black)
[![Checked with mypy](http://www.mypy-lang.org/static/mypy_badge.svg)](http://mypy-lang.org/)
[![license](https://img.shields.io/badge/License-Apache_2.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)

> 

Author: Beautiful! Thanks!

# OpenSquirrel

[![CI](https://github.com/QuTech-Delft/OpenSquirrel/workflows/Tests/badge.svg)](https://github.com/qutech-delft/OpenSquirrel/actions)

> [![PyPI](https://badgen.net/pypi/v/OpenSquirrel)](https://pypi.org/project/OpenSquirrel/)

Author: I'm working on this for libqasm. I'll do something similar for OpenSquirrel, yes.

- Python versions, I'll have a look and add it.
- Code coverage, when we have it in the future.

![image](https://github.com/user-attachments/assets/b81b0ecf-99cc-46e1-bd07-b7b907a66355)

> [![PyPI](https://badgen.net/pypi/v/OpenSquirrel)](https://pypi.org/project/OpenSquirrel/)

Author: I've added a few to OpenSquirrel.

![image](https://github.com/user-attachments/assets/3ab66c11-d70d-46cc-8843-c01567dca476)

But code coverage has to be set up in `codecov` for the badge to work.

It understands the quantum programming language cQASM 3 and will support additional quantum programming languages in the future.

> It is developed in modern Python and follows best practices.

Author: I'm leaving it like: `It supports the cQASM quantum programming language, using [libQASM](https://github.com/QuTech-Delft/libqasm) as language parser.`

## User manual

> OpenSquirrel documentation is hosted through GitHub Pages [here](https://QuTech-Delft.github.io/OpenSquirrel/).

Reviewer: ```suggestion
[User documentation of OpenSquirrel](https://QuTech-Delft.github.io/OpenSquirrel/) is hosted through GitHub Pages.
```

Author: I'm leaving it like: `User documentation of OpenSquirrel is hosted through [GitHub Pages](https://QuTech-Delft.github.io/OpenSquirrel/).`

Reviewer: Wouldn't be more clear to the user to link from 'User documentation of OpenSquirrel' instead of linking from 'GitHub pages', the latter is the means by which we publish, and the former is actually what the link points to.

Author: Done!

## File organization

For development, see:

- `opensquirrel`: source files.
- `test`: test files.

For build process, continuous integration, and documentation:

- `.github`: GitHub Actions configuration files.
- `docs`: documentation files.
- `scripts`: documentation helper script.

> ## Dependencies

Reviewer: Has this been added to the README?

> ## Dependencies

Author: We have a Dependencies section and an Installation section now, yes. The Installation from source is just a clone, I think it belongs to the GitHub Pages, Developer guide section.

> All dependencies are managed via `poetry`.

Reviewer: ```suggestion
All dependencies are managed via [Poetry](https://python-poetry.org/).
```

Author: I'm leaving it like: `All dependencies are managed via [poetry](https://python-poetry.org/).`.

From an OpenSquirrel checkout:

```shell
poetry shell
poetry install
```

## Use from another project

> The `opensquirrel` module can be imported from another Python file with:

Reviewer: ```suggestion
To use OpenSquirrel, the `opensquirrel` module can be imported within Python accordingly:
```

Author: Too complicated.

Author: OK, I've left it like:

> The `opensquirrel` module can be imported from Python:

```python
import opensquirrel
```

> 

Reviewer: We should add a Contributing section detailing how contributing developers can get started and how they can run linting, mypy, and tests.
